README.md

```diff
@@ -1,210 +1,440 @@
-- Support list cached connections to foreign servers by using function griddb_get_connection().
-- Support discard cached connections to foreign servers by using function griddb_disconnect('server_name'), griddb_disconnect_all().
-- Support deparse and push down ANY/ALL ARRAY with the argument is constant.
+```sql
+	GRANT USAGE ON FOREIGN SERVER sqlite_server TO pguser;
+```
+Where `pguser` is a sample user for works with foreign server (and foreign tables).
+
+### User mapping
+
+Create an appropriate user mapping:
+```sql
+    	CREATE USER MAPPING
+	FOR pguser
+	SERVER griddb_svr 
+    	OPTIONS (
+	  username 'username',
+	  password 'password'
+	);
+```
+Where `pguser` is a sample user for works with foreign server (and foreign tables).
+
+### Create foreign table
+All `CREATE FOREIGN TABLE` SQL commands can be executed as a normal PostgreSQL user if there were correct `GRANT USAGE ON FOREIGN SERVER`. No need PostgreSQL supersuer for security reasons but also works with PostgreSQL supersuer.
+
+Create a foreign table referencing the griddb table `fdw_test`:
+```sql
+	CREATE FOREIGN TABLE ft1 (
+	  c1 text,
+	  c2 float,
+	  c3 integer
+	)
+	SERVER griddb_svr;
+```
+Query the foreign table.
+```sql
+	SELECT * FROM ft1;
+```
+The **container must have rowkey on GridDB in order to execute update and delete query**.	
 
-## 4. Limitations
-#### Record is updated by INSERT command if a record with same rowkey as new record exists in GridDB.
-In this case, griddb_fdw raises the warning.
+### Import a GridDB schema:
 
+```sql
+	IMPORT FOREIGN SCHEMA public
+	FROM SERVER griddb_svr
+	INTO public
+	OPTIONS (
+	  recreate 'true'
+	);
 ```
+
+After schema is imported, we can access tables.
+To use `CREATE FOREIGN TABLE` is not recommended.
+
+Limitations
+-----------
+### SQL commands
+- Record is updated by `INSERT` command if a record with same rowkey as new record exists in GridDB.
+In this case, `griddb_fdw` raises the warning.
+
+```sql
 INSERT INTO ft1 VALUES(100, 'AAA');
 INSERT INTO ft1 VALUES(100, 'BBB'); -- Same as "UPDATE ft1 SET b = 'BBB' WHERE a = 100;"
 ```
+- Don't support `ON CONFLICT`. PostgreSQL has upsert (update or insert) feature. When inserting a new row into the table, PostgreSQL will update the row if it already exists, otherwise, PostgreSQL inserts the new row. However, `griddb_fdw` does not support the upsert feature now. For example, it shows an error message if user executes the following query.
+
+```sql
+INSERT INTO ft1(c1, c2) VALUES(11, 12) ON CONFLICT DO NOTHING;
+```
+
+- Don't support `RETURNING`. Returning is way to obtain data if rows are modified by `INSERT`, `UPDATE`, and `DELETE` commands. `griddb_fdw` does not support this feature.
+```sql
+INSERT INTO ft1 (c0, c1) VALUES (1, 2) RETURNING c0, c1;
+```
+- Don't support DIRECT MODIFICATION
+- Don't support `IMPORT FOREIGN SCHEMA` with option `import_generated` or `LIMIT TO` clause
```

Comment:
This description seems be not correct:
- Griddb FDW makes CREATE FOREIGN TABLE statements for all existed tables in remote server, and then PostgreSQL's core checks LIMIT TO or EXCEPT conditions.
- `import_generated` is an option of postges_fdw and it has no meaning with griddb_fdw.

So, could you update this information?

I think the description should be:
```suggestion
- Don't support push down LIMIT TO | EXCEPT filter in IMPORT FOREIGN SCHEMA. These filters will run locally.
```

Reply:
Thanks @nxhai98  , changed.

Reply:
Thank you for fixing,